smpplib/client.py

```diff
@@ -194,13 +201,25 @@ def send_pdu(self, p):
                 raise exceptions.ConnectionError()
             sent += sent_last
 
+        if p.command[-5:] == '_resp':
+            self.outstanding_operations -= 1
+        else:
+            self.outstanding_operations += 1
+
         return True
 
     def read_pdu(self):
         """Read PDU from the SMSC"""
 
         self.logger.debug('Waiting for PDU...')
 
+        if pdu.command[-5:] == '_resp':
+            self.outstanding_operations -= 1
+            if self.outstanding_operations < 0:
+                self.logger.warning('Number of outstanding operations < 0')
+        else:
+            self.outstanding_operations += 1
```

Comment:
Mmm, I don't really get this part. Why reading a PDU really increases the number?

Reply:
In this case we are reading a PDU which isn't a response so it's the SMSC asking for something, so it is an outstanding operation until we actually answer it.